packages/@react-aria/overlays/src/usePopover.ts

```diff
@@ -82,7 +82,8 @@ export function usePopover(props: AriaPopoverProps, state: OverlayTriggerState):
 
   let {overlayProps, underlayProps} = useOverlay(
     {
-      isOpen: state.isOpen,
+      // If popover is in the top layer, it should not prevent other popovers from being dismissed.
+      isOpen: state.isOpen && !otherProps['data-react-aria-top-layer'],
```

Comment:
This is kinda not ideal. Should we have a more explicit prop for this? Basically we have to prevent submenus from being added to the `visibleOverlays` stack inside `useOverlay`, that way they don't prevent the click outside behavior of the root menu.

Reply:
For posterity, we discussed this and decided to hold off on making a separate prop until after release, will need to think more about the case where this will cause a submenu to still be visible/interactable when a separate modal opens (something that we do want for Toasts via top-layer but not for submenus)